Framework/Algorithms/test/Stitch1DManyTest.h

```diff
@@ -147,182 +153,175 @@ class Stitch1DManyTest : public CxxTest::TestSuite {
   }
 
   void test_throws_with_too_few_workspaces() {
+    createUniformWorkspace(0.1, 0.1, 1., 2., "ws1");
     Stitch1DMany alg;
     alg.setChild(true);
     alg.initialize();
     alg.setProperty("InputWorkspaces", "ws1");
     alg.setProperty("Params", "0.1, 0.1, 1.8");
-    alg.setPropertyValue("OutputWorkspace", "outws");
+    alg.setProperty("OutputWorkspace", "outws");
     TS_ASSERT_THROWS(alg.execute(), std::runtime_error);
```

Comment:
**Good point**, I thought about it earlier (sorry for the long answer):

This was the intention to be more clear in `ConjoinXRunsTest` but I am unhappy now about this test: it tests input validation that always throws `std::runtime_error` with message `Some invalid properties found`. Let's say less informative?

To the contrary, if the algorithm throws, it is in my opinion important to test if exactly this error throws with the expected message. That is why I think in most code input validation messages do not get checked (?). On the other hand side, we can be really clear that input validation throws and nothing else -> what else can throw here? If the algorithm itself does not throw `std::runtime_error`s there is less possibility and it can only be input validation? However, `Stitch1D` throws `std:runtime_errors`.

Finally, `Some invalid properties found` does not tell us if it really is the `too_few_workspaces` thing the unit test wants to test. (Can this be improved?). Only the developer knows this.

Do you think I should do `TS_ASSERT_THROWS_EQUALS`? This influences all my future testing ...

Reply:
If you just get `Some invalid properties found` I agree the extra test is less useful. It still tells you something - that the validation failed and not, say, something later in the execution.

I think it should be your decision if you think it is useful to add the extra asserts. Perhaps not here, but for the `Stitch1DTests` which use `RunCombinationHelpers`. If `TS_ASSERT_THROWS` is true then there should be no serious problems, but I have seen cases where an algorithm throws for an unexpected reason - which means unexpected behaviour in the code somewhere and a less helpful message to the user.